contracts/MetadataViews.cdc

```diff
@@ -11,113 +12,38 @@ import NonFungibleToken from "NonFungibleToken"
-        pub let uuid: UInt64
-        pub let display: Display?
-        pub let externalURL: ExternalURL?
-        pub let collectionData: NFTCollectionData?
-        pub let collectionDisplay: NFTCollectionDisplay?
-        pub let royalties: Royalties?
-        pub let traits: Traits?
-
-        init(
-            id : UInt64,
-            uuid : UInt64,
-            display : Display?,
-            externalURL : ExternalURL?,
-            collectionData : NFTCollectionData?,
-            collectionDisplay : NFTCollectionDisplay?,
-            royalties : Royalties?,
-            traits: Traits?
-        ) {
-            self.id = id
-            self.uuid = uuid
-            self.display = display
-            self.externalURL = externalURL
-            self.collectionData = collectionData
-            self.collectionDisplay = collectionDisplay
-            self.royalties = royalties
-            self.traits = traits
-        }
-    }
-
-    /// Helper to get an NFT view
-    ///
-    /// @param id: The NFT id
-    /// @param viewResolver: A reference to the resolver resource
-    /// @return A NFTView struct
-    ///
-    pub fun getNFTView(id: UInt64, viewResolver: &{Resolver}) : NFTView {
-        let nftView = viewResolver.resolveView(Type<NFTView>())
-        if nftView != nil {
-            return nftView! as! NFTView
-        }
-
-        return NFTView(
-            id : id,
-            uuid: viewResolver.uuid,
-            display: self.getDisplay(viewResolver),
-            externalURL : self.getExternalURL(viewResolver),
-            collectionData : self.getNFTCollectionData(viewResolver),
-            collectionDisplay : self.getNFTCollectionDisplay(viewResolver),
-            royalties : self.getRoyalties(viewResolver),
-            traits : self.getTraits(viewResolver)
-        )
-    }
+access(all) contract MetadataViews {
 
     /// Display is a basic view that includes the name, description and
     /// thumbnail for an object. Most objects should implement this view.
     ///
-    pub struct Display {
+    access(all) struct Display {
 
         /// The name of the object.
         ///
         /// This field will be displayed in lists and therefore should
         /// be short an concise.
         ///
-        pub let name: String
+        access(all) let name: String
 
         /// A written description of the object.
         ///
         /// This field will be displayed in a detailed view of the object,
         /// so can be more verbose (e.g. a paragraph instead of a single line).
         ///
-        pub let description: String
+        access(all) let description: String
 
         /// A small thumbnail representation of the object.
         ///
         /// This field should be a web-friendly file (i.e JPEG, PNG)
         /// that can be displayed in lists, link previews, etc.
         ///
-        pub let thumbnail: AnyStruct{File}
+        access(all) let thumbnail: {File}
 
-        init(
+        view init(
```

Comment:
Curious how come this is `view` but also changing state values? I see this pattern in other places, so for sure it's something I don't know about.

Reply:
It can be `view` because it isn't changing any existing state. It is a newly created struct, so it is basically treated like a getter function

Reply:
I agree it isn't super intuitive at a first glance though